src/Backups/BackupInfo.cpp

```diff
@@ -98,4 +99,29 @@ String BackupInfo::toStringForLogging() const
     return toAST()->formatForLogging();
 }
 
+void BackupInfo::copyS3CredentialsTo(BackupInfo & dest) const
+{
+    /// named_collection case, no need to update
+    if (!dest.id_arg.empty() || !id_arg.empty())
+        throw Exception(ErrorCodes::BAD_ARGUMENTS, "use_same_s3_credentials_for_base_backup is not compatible with named_collections");
+
+    /// All of this is invalid arguments, and copyS3CredentialsTo() should not
+    /// be called in this case, since it should fail in
+    /// registerBackupEngineS3(), hence LOGICAL_ERROR
+    if (backup_engine_name != "S3")
+        throw Exception(ErrorCodes::LOGICAL_ERROR, "use_same_s3_credentials_for_base_backup supported only for S3, got {}", toStringForLogging());
+    if (dest.backup_engine_name != "S3")
+        throw Exception(ErrorCodes::LOGICAL_ERROR, "use_same_s3_credentials_for_base_backup supported only for S3, got {}", dest.toStringForLogging());
```

Comment:
Again, why is it logical error? If one executes

```
BACKUP ... TO Disk(...) SETTINGS use_same_s3_credentials_for_base_backup = true
```

it's an error, but we should only tell the user that it's an error, not immediately stop the server.

Reply:
Yep, you are right about checks for `backup_engine_name`. Nice catch. Fixed (and updated the test as well).